biorxiv/exploratory_data_analysis/EDA_BioRxiv.py

```diff
@@ -0,0 +1,262 @@
+    # why is there a <sc> tag?
+    if any(list(map(lambda x: len(x) > 1, title_objs))):
+    
+        # filter out weird characters ⓘ
+        # cant think of a better way to handle these types of edge cases
+        title_objs = list(map(lambda headers: list(filter(lambda token: token != 'ⓘ', headers)), title_objs))
+        title_objs = list(map(lambda x: x[0] + x[1] if len(x) > 1 else x, title_objs))
+    
+    abstract_section = root.xpath('//abstract/title//text()')
+    if len(abstract_section) > 0:
+        
+        # in case of a parse error that splits A from bstract
+        if len(abstract_section) > 1:
+            abstract_section = ["".join(abstract_section)]
+        
+        title_objs = title_objs + [abstract_section]
+
+    title_objs = list(map(lambda x: x[0].rstrip().lower(), title_objs))
+    
+    article_sections += list(
+        map(
+            lambda x: {'section':header_group_mapper(x[0]), 'document':x[1]},
+            product(title_objs, [article])
+        )
+    )
+
+
+# In[5]:
+
+
+metadata_df = (
+    pd.DataFrame
+    .from_records(article_metadata)
+    .fillna({"category":'none', 'author_type':'none', 'heading':'none'})
+    .assign(category=lambda x:x.category.apply(lambda x: " ".join(x.split("_")) if "_" in x else x))
+)
+
+metadata_df.to_csv("output/biorxiv_article_metadata.tsv", sep="\t", index=False)
+metadata_df.head()
+
+
+# In[6]:
+
+
+sections_df = (
+    pd.DataFrame
+    .from_records(article_sections)
+)
+
+sections_df.to_csv("output/biorxiv_article_sections.tsv", sep="\t", index=False)
+sections_df.head()
+
+
+# # Regular Research Articles?
+
+# BioRxiv claims that each article should be a research article. The plot below mainly confirms that statement.
+
+# In[7]:
+
+
+g = (
+    p9.ggplot(metadata_df, p9.aes(x="author_type"))
+    + p9.geom_bar(size=10)
+    + p9.theme_bw()
+)
+print(g)
+
+
+# In[8]:
+
+
+metadata_df["author_type"].value_counts()
+
+
+# # BioRxiv Research Article Categories
+
+# Categories assigned to each research article. Neuroscience dominates majority of the articles as expected.
+
+# In[9]:
+
+
+category_list = metadata_df.category.value_counts().index.tolist()[::-1]
+
+# plot nine doesn't implement reverse keyword for scale x discrete
+# ugh...
+g = (
+    p9.ggplot(metadata_df, p9.aes(x="category"))
+    + p9.geom_bar(size=10)
+    + p9.scale_x_discrete(limits=category_list)
+    + p9.coord_flip()
+    + p9.theme_bw()
+)
+g.save("output/figures/preprint_category.png", dpi=500)
+print(g)
+
+
+# In[10]:
+
+
+metadata_df["category"].value_counts()
+
+
+# # New, Confirmatory, Contradictory Results?
+
+# In[11]:
+
+
+heading_list = metadata_df.heading.value_counts().index.tolist()[::-1]
+
+g = (
+    p9.ggplot(
+        metadata_df
+        .rename({
+        "bioinformatics":"none",
+        "genomics": "none",
+        "zoology": "none",
+        "evolutionary_biology": "none",
+        "animal behavior and cognition": "none",
+        "ecology":"none"
+        }), 
```

Comment:
Are article categories something you expect to see in this field? Is it possible that the metadata for the heading is missing, causing the parser to skip the header column? If so, will this disrupt downstream analyses?

Reply:
Good catch. When I was analyze this dataset I noted that some of the documents had errors where the document's heading section was replaced with the document categories. I quantified how many documents had this error and it is a negligible amount (15 out of 98k documents). For ease of analysis and maintaining sanity I decided to group these errors into the none category.